src/bigint.rs

```diff
@@ -111,12 +111,28 @@ impl<'de> serde::Deserialize<'de> for Sign {
 }
 
 /// A big signed integer type.
-#[derive(Clone, Debug, Hash)]
+#[derive(Debug, Hash)]
 pub struct BigInt {
     sign: Sign,
     data: BigUint,
 }
 
+impl Clone for BigInt {
```

Comment:
What does this offer over deriving `Clone`? Sign is not `Clone`'d in this implementation, but it is `Copy`'d, and I believe those will compile down to the same thing, so I don't see how this affects runtime.

Reply:
The derived `Clone` doesn't specialize `clone_from`. `Vec` can reuse its allocation through that, but the default `Clone::clone_from` just clobbers the old value with a whole new `clone`. I'll add a comment.